### ISSUE TITLE
[frio] Screen Reader Accessibility Part 6: Conversation profile photo presentation

### ISSUE THREAD

**Issue by MrPetovan** [friendica/friendica#10981]
@pvagner
> When looking at the conversation threads I am not sure but I guess there is a potential issue with presentation of user photo. When navigating with the tab key there is a link, it has an user photo as its content and the alt property of that image includes display name of the user. The link points to user profile page. Then there is link to source. And also there is another link to the profile page with clear tooltip. I am not sure if I am understanding this right but I think the presentation would be more stream lined for screen reader users if the link with profile photo would not be focusable and would be hidden from the screen reader users. There might be a situation when profile photo does something else than the surrounding text and in that case the link target might be more important than the presentation for the screen reader users. Can you think about this a bit please?

Source: https://github.com/friendica/friendica/issues/1743#issuecomment-961800209

**Comment by pvagner** [comment]
Given the current social media changes when activitypub based apps are getting influx of a new users it also includes more screen reader users interested.
I understand from a technical standpoint contributing to this issue may not be difficult.
However an advice from people knowing friendica would be appreciated and well received here. I am afraid by hiding the photo from screen reader users I may inadvertently hide some functionality with it. There is supposed to be an contact hover menu somewhere and I am not sure but I think I am unable to access that one with the keyboard alone. Might that be related to this please?

**Comment by MrPetovan** [comment]
You are correct, on the profile picture there is a link that when clicked/tapped, shows this contact hover menu. This menu includes the following elements:
- Profile picture
- Display name
- Account address
- Account hashtags
- Link to the contact conversations
- Link to Follow/Unfollow
- Link to start a private conversation if any
- Link to the local contact page

This hover menu also shows up when hovering with the mouse on the display name in the regular post structure. Since it wasn't accessible to mobile touchscreen, I added this click-based interaction on the profile picture to show it.

**Comment by pvagner** [comment]
Well, I really can't locate this menu. I am even unsure if it can be activated by using the keyboard alone.
Am I looking at the correct link and image?
Is it the first focusable link when navigating to a post using the tab key?
Where in the DOM tree is the menu? Near the image or at the end?

**Comment by MrPetovan** [comment]
Here's the abridged DOM from the start of the post to the link I'm talking about:
```
<div class="item- wall-item-container  apub thread_level_1 panel-body h-entry" id="item-">
	<div class="media shiny">
		<div class="dropdown pull-left">
			<div class="hidden-sm hidden-xs contact-photo-wrapper mframe p-author h-card">
				<a class="userinfo click-card u-url" id="wall-item-photo-menu-" href="https://octodon.social/users/kingu_platypus_gidora">
					<div class="contact-photo-image-wrapper">
						<img src="photo/contact/80/10736?ts=" class="contact-photo media-object  p-name u-photo" id="wall-item-photo-" alt="Kingu Platypus">
					</div>
				</a>
			</div>
		</div>
		<div role="heading" aria-level="1">
...
```

**Comment by pvagner** [comment]
Yes, that's the link with an image I am looking at as well.
The link target is remote URL and there is no onclick handler. What is used to show a menu and where is that menu generated?
Is it generated dynamically? Most other menus are generated as a part of the DOM and shown / hidden according to the input events. So I have difficulties identifying this one.

**Comment by MrPetovan** [comment]
The onclick event is added in javascript, and it shows a hovercard which contents is loaded asynchronously depending on the contact being interacted with.

I feel like for screen reader use, enabling the "Stay Local" settings in Settings -> Display -> Content/Layout is a must, it would transform these external links to local links to the local copy of the contact information. This page has all the information and actions the hovercard has.
